study-group/Toast-Implementation-Studies.md

```diff
@@ -70,25 +70,190 @@ The `<paper-toast>` holds onto information about how it was dismissed the last t
+The `text`, `view`, `gravity`, and `duration` of the toast are all the elements the developer can control when working with this very straightforward component.
+
+
+### Notable Features & Details
+
+#### Gravity
+The library takes the Android approach to positioning,
+by letting the user define the Gravity of the element,
+along with offset values for the axes.
+
+#### Duration Constants
+The library comes with baked in constants for displaying the toast for long and short periods of time.
+
+### Takeaways
+- High opinion, low control.
+- The API for the `Toast` object provides getters and setters for each of `text`, `view`, `gravity`, `offset` and `duration`.
+- Beyond getters and setters, the only control the object allows is `.makeText()`, `.show()` and `.cancel()`.
+
+## [react-toastify](https://github.com/fkhadra/react-toastify)
+`react-toastify` is a React component library by author [@fkhadra](https://github.com/fkhadra) for rich, extensible notification toasts.
+
+### Design & Sample Code
+
+To use a `react-toastify` toast requires some simple JavaScript:
+
+```js
+class App extends Component {
+  notify = () => toast("Very simple toast!");
+
+  render(){
+    return (
+      <div>
+        <button onClick={this.notify}>Notify !</button>
+        <ToastContainer />
+      </div>
+    );
+  }
+}
+```
+
+`react-toastify` uses the explicit component pattern, by having the developer render a React `ToastContainer` component (the recommendation is to put this in the application root).
+This single component will render all toasts called from JavaScript and can be configured with global options.
+However, the library allows the developer a lot of freedom, supporting multiple `ToastContainer`s as well as no `ToastContainer`, though one is recommended.
+
+When the `toast()` is called with a message and a props object, it is displayed in the toast container, with the toast props overriding the `ToastContainer` props, like so:
+
+```js
+class App extends Component {
+  // Creating a toast that will close in 10000ms.
+  tenSecondToast = () => toast("This toast will self destruct in 10...", { autoClose: 10000 });
+  // Creating a toast that will close in 5000ms (the default).
+  normalToast = () => toast("This is a normal toast");
+
+  render(){
+    return (
+      <div>
+        <button onClick={this.tenSecondToast}>Close after 10000ms seconds</button>
+        <button onClick={this.normalToast}>Close after 5000ms (default)</button>
+        <ToastContainer autoClose={5000} />
+      </div>
+    );
+  }
+}
+```
+
+For some of the common controls, the toast can be configured like so:
+
+```js
+class App extends Component {
+  topRightWarning = () => toast('Warning!', {
+    type: toast.TYPE.WARNING,
+    position: toast.POSITION.TOP_RIGHT,
+    onClose: ({ props }) => window.alert('I warned you') // props: all props passed to rendered toast component
```

Comment:
Maybe just omit the props argument since it's not essential to the example? i.e. `() => window.alert(...)`

Reply:
I think I can find a better way to include them
```suggestion
    onClose: ({ props }) => window.alert(`You just closed the ${props.type}`) // props: all props passed to rendered toast component
```